src/peergos/server/tests/simulation/Simulator.java

```diff
@@ -263,13 +270,18 @@ public Simulator(int opCount, Random random, long meanFileLength,
         this.index = new FileSystemIndex(random);
     }
 
-    private boolean read(String user, Path path) {
+    private void readFile(String user, Path path) {
         FileSystem testFileSystem = fileSystems.getTestFileSystem(user);
         FileSystem referenceFileSystem = fileSystems.getReferenceFileSystem(user);
+        referenceFileSystem.read(path);
+        testFileSystem.read(path);
+    }
 
-        byte[] refBytes = referenceFileSystem.read(path);
-        byte[] testBytes = testFileSystem.read(path);
-        return Arrays.equals(refBytes, testBytes);
+    private void readDirectory(String user, Path path) {
```

Comment:
1.  I think the concept of "reading" a  directory is a bit confusing  and doesn't directly map onto (is it read & execute?) - why introduce it?
2. What is this supposed to check?  Possible side effects of  `ls`-ing a  directory?

Reply:
1. read == list. we use the word read for file so it seemed appropriate. we have write directory for make directory.
2. yes, the code can throw exceptions under a number of circumstances.